### PR TITLE
Reduce memory fragment

### PR DESCRIPTION
**Pull request by w-zr**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #16653

## What this PR does / why we need it:

create rowid from objID, blkNum and offset at once.

### CODE REVIEW

File: pkg/container/types/rowid.go [hunk 1]
```diff
@@ -74,6 +74,15 @@ func NewRowid(blkid *Blockid, offset uint32) *Rowid {
 	return &rowid
 }
 
+func NewRowIDWithObjectIDAndBlockNum(segid *Objectid, blknum uint16, offset uint32) *Rowid {
```
Comment (fengttt) [inline pkg/container/types/rowid.go:77]: Why we cannot just return the Rowid?   There is no need for pointers.

**Review by fengttt — CHANGES_REQUESTED** [review]

**Review by fengttt — CHANGES_REQUESTED** [review]
Id is only a few bytes, there is no need for pointer at all.

**Review by fengttt — COMMENTED** [review]
api.proto missing?

File: pkg/container/types/rowid.go (hunk 1 above)
Comment (w-zr) [reply pkg/container/types/rowid.go:77]: Should we represent all RowID, BlockID, and ObjectID with values, or just modify this function?

**Comment by w-zr** [comment]
> api.proto missing?

I will remove api.proto change in this pr.

File: pkg/container/types/rowid.go (hunk 1 above)
Comment (w-zr) [reply pkg/container/types/rowid.go]: I modified the Objectid parameter and return type into value.

**Review by fengttt — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

**Comment by w-zr** [comment]
dup with https://github.com/matrixorigin/matrixone/pull/17768

**Pull request closed without merging**